classes/announcement/Announcement.php

```diff
@@ -1,345 +1,430 @@
-     * Get assoc type for this announcement.
-     *
-     * @return int
-     */
-    public function getAssocType()
-    {
-        return $this->getData('assocType');
-    }
+    protected $table = 'announcements';
+    protected $primaryKey = 'announcement_id';
+    public const CREATED_AT = 'date_posted';
+    public const UPDATED_AT = null;
+    protected string $settingsTable = 'announcement_settings';
 
     /**
-     * Set assoc type for this announcement.
+     * The attributes that are mass assignable.
      *
-     * @param int $assocType
+     * @var array
      */
-    public function setAssocType($assocType)
-    {
-        $this->setData('assocType', $assocType);
-    }
+    protected $fillable = ['assocType', 'assocId', 'typeId', 'title', 'image', 'description', 'descriptionShort'];
 
     /**
-     * Get the announcement type of the announcement.
-     *
-     * @return int
+     * @inheritDoc
      */
-    public function getTypeId()
+    public static function getSchemaName(): string
     {
-        return $this->getData('typeId');
+        return PKPSchemaService::SCHEMA_ANNOUNCEMENT;
     }
 
     /**
-     * Set the announcement type of the announcement.
-     *
-     * @param int $typeId
+     * @inheritDoc
      */
-    public function setTypeId($typeId)
+    public function getSettingsTable(): string
     {
-        $this->setData('typeId', $typeId);
+        return $this->settingsTable;
     }
 
     /**
-     * Get the announcement type name of the announcement.
-     *
-     * @return string|null
+     * Add Model-level defined multilingual properties
      */
-    public function getAnnouncementTypeName()
+    public function getMultilingualProps(): array
     {
-        $announcementTypeDao = DAORegistry::getDAO('AnnouncementTypeDAO'); /** @var AnnouncementTypeDAO $announcementTypeDao */
-        return $this->getData('typeId') ? $announcementTypeDao->getById($this->getData('typeId'))?->getLocalizedTypeName() : null;
+        return array_merge(
+            $this->multilingualProps,
+            [
+                'fullTitle',
+            ]
+        );
     }
 
     /**
-     * Get localized announcement title
+     * Delete announcement, also allows to delete multiple announcements by IDs at once with destroy() method
      *
-     * @return string
-     */
-    public function getLocalizedTitle()
-    {
-        return $this->getLocalizedData('title');
-    }
-
-    /**
-     * Get full localized announcement title including type name
+     * @return bool|null
      *
-     * @return string
+     * @hook Announcement::delete::before [[$this]]
      */
-    public function getLocalizedTitleFull()
+    public function delete()
     {
-        $typeName = $this->getAnnouncementTypeName();
-        if (!empty($typeName)) {
-            return $typeName . ': ' . $this->getLocalizedTitle();
-        } else {
-            return $this->getLocalizedTitle();
+        Hook::call('Announcement::delete::before', [$this]);
```

Comment:
We use to have two relevant hooks here: `Announcement::delete::before` (still here, but will require callers to change from expecting a `DataObject` to expecting an Eloquent model) and `Announcement::delete' (now removed).

Big picture: I'd suggest we rely on Eloquent lifecycle events for entities that use Eloquent. This will reduce the ambiguity over whether to use these vs. hooks, and will standardize us towards Laravel more generally.

Smaller picture: Notwithstanding the above, when we change hook conventions...
- If a hook is removed, make sure to add a call to `Hook::addUnsupportedHooks` e.g. in `PKPApplication` so that registrants are prevented from trying to use it. (See [#10163](https://github.com/pkp/pkp-lib/issues/10163))
- If a hook is kept but substantially changed, consider [adopting the newer [`Hook::run` conventions](https://github.com/pkp/pkp-lib/discussions/8089) since we're already introducing a breaking change, and noting it in the list of changes for the [3.5 release notebook](https://github.com/pkp/pkp-lib/issues/9276). In particular, `Hook::run` allows for us to use typehinting and named parameters instead of the older clumsy array-based parameter passing.